progress/__init__.py

```diff
@@ -65,19 +65,20 @@ def start(self):
     def finish(self):
         pass
 
-    def next(self, n=1):
-        if n > 0:
-            now = time()
-            dt = now - self._ts
-            if dt < self.time_threshold:
-                self._pending += n
-            else:
-                self._xput.append((n + self._pending) / dt)
-                self._ts = now
-                self._pending = 0
 
+    def next(self, n=1):
+        if n <= 0: return
+        now = time()
+        dt = now - self._ts
+        self._pending += n
+        #avoid performing computationally intensive tasks more than 1/dt (10 by default) times a second
+        if dt > self.time_threshold:
+            self._xput.append((self._pending) / dt)
+            self._ts = now
+            self._pending = 0
+            self.update()
         self.index = self.index + n
-        self.update()
+
 
```

Comment:
There's no logic to #2 - the motivation is purely stylistic.  I think you are right - there is one extra assignment operation incurred by my method.